core/src/main/java/org/bitcoinj/script/Script.java

```diff
@@ -393,16 +395,17 @@ public Address getToAddress(NetworkParameters params) throws ScriptException {
      *            showing addresses rather than pubkeys.
      */
     public Address getToAddress(NetworkParameters params, boolean forcePayToPubKey) throws ScriptException {
+        Network network = params.network();
         if (ScriptPattern.isP2PKH(this))
-            return LegacyAddress.fromPubKeyHash(params, ScriptPattern.extractHashFromP2PKH(this));
+            return LegacyAddress.fromPubKeyHash(network, ScriptPattern.extractHashFromP2PKH(this));
         else if (ScriptPattern.isP2SH(this))
-            return LegacyAddress.fromScriptHash(params, ScriptPattern.extractHashFromP2SH(this));
+            return LegacyAddress.fromScriptHash(network, ScriptPattern.extractHashFromP2SH(this));
         else if (forcePayToPubKey && ScriptPattern.isP2PK(this))
-            return LegacyAddress.fromKey(params, ECKey.fromPublicOnly(ScriptPattern.extractKeyFromP2PK(this)));
+            return LegacyAddress.fromPubKeyHash(network, ECKey.fromPublicOnly(ScriptPattern.extractKeyFromP2PK(this)).getPubKeyHash());
```

Comment:
Just noticed this change, is it correct?

Reply:
Yes, the functionality is equivalent. Since `fromKey` was deprecated, `fromPubKeyHash` achieves the same result. I think it would also be possible to use `ECKey.toAddress()` but using a `static` method of `LegacyAddress` matched the other cases more closely.

Reply:
To clarify, this switches from `fromKey` to `fromPubKeyHash`.

Reply:
Ok thanks!